### PR TITLE
definition for splitpanes version 2.2

### PR DESCRIPTION
**Pull request by N0K0**
- [ x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ x ] Test the change in your own code. (Compile and run.)
- [ x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ x ] Provide a URL to documentation or source code which provides context for the suggested changes: Missing definition after breaking changes in atleast 2.0. https://github.com/antoniandre/splitpanes 
- [ x ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ x ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ x ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Hopefully this is correct! :)
Been using the other Vue type projects as reference

### CODE REVIEW

**Comment by N0K0** [comment]
Now i feel silly, my **npm run test** was just a name.. Not what it did in Webstorm. Sorry about that..

**Comment by N0K0** [comment]
Bump, given the Abandoned tag

File: types/splitpanes/index.d.ts [hunk 1]
```diff
@@ -1,85 +1,198 @@
-// Type definitions for splitPanes 1.0
+// Type definitions for splitPanes 2.2
 // Project: https://github.com/antoniandre/splitpanes
-// Definitions by: teenhe <https://github.com/noonhorse>
+// Definitions by: n0k0 <https://github.com/n0k0>
```
Comment (andrewbranch) [inline types/splitpanes/index.d.ts:3]: Did @noonhorse ask to be removed as an owner?

Comment (N0K0) [reply types/splitpanes/index.d.ts:3]: Hi!
No, but this is a complete rewrite of a definition that seemed abandoned.

**Comment by N0K0** [comment]
I think we maybe should close this PR and prioritize adding the correct typing directly to the project. 
Ref: https://github.com/antoniandre/splitpanes/issues/82

**Comment by andrewbranch** [comment]
Why? This is the place where type definitions for projects not written in TypeScript go. It sounds like the author has no interest in managing the type definitions, so it doesn’t make sense for them to go in the source.

File: types/splitpanes/index.d.ts (hunk 1 above)
Comment (andrewbranch) [reply types/splitpanes/index.d.ts:3]: That's fair 👍

File: types/splitpanes/index.d.ts [hunk 2]
```diff
@@ -1,85 +1,198 @@
...
+
+export const Pane: Pane;
+export const Splitpane: Splitpane;
+
+declare let _default: {
+    Pane: Pane,
+    Splitpane: Splitpane,
+};
+
+export default _default;
```
Comment (andrewbranch) [inline types/splitpanes/index.d.ts:198]: I don’t actually see a default export in the source JS.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/splitpanes/index.d.ts (hunk 2 above)
Comment (N0K0) [reply types/splitpanes/index.d.ts:198]: Sorry for the delay! Will look into this this weekend! :)

Comment (N0K0) [reply types/splitpanes/index.d.ts:198]: I'm a bit stumped, what would be the correct way to export both `Pane` and `SplitPanes`?

Comment (andrewbranch) [reply types/splitpanes/index.d.ts:198]: You’ve already done it:

```ts
export const Pane: Pane;
export const Splitpane: Splitpane;
```

Although now that I look at it again, it appears the exports are named `Pane` and `Splitpanes` (with a trailing `s`), whereas your second export is missing that trailing `s`.

File: types/splitpanes/index.d.ts [hunk 3]
```diff
@@ -1,85 +1,191 @@
...
+    computed: PaneComputed;
+    watch: PaneWatch;
+    mounted: () => void;
+    beforeDestroy: () => void;
+    beforeUpdate: () => void;
+    render: (createEl: any) => any;
+}
+
+export const Pane: Pane;
+export const Splitpane: Splitpane;
```
Comment (andrewbranch) [inline types/splitpanes/index.d.ts:191]: The export is named `Splitpanes`. See https://unpkg.com/browse/splitpanes@2.3.6/src/components/splitpanes/index.js.

**Review by andrewbranch — CHANGES_REQUESTED** [review]

File: types/splitpanes/index.d.ts (hunk 3 above)
Comment (N0K0) [reply types/splitpanes/index.d.ts]: Updated!

**Comment by andrewbranch** [comment]
Import name just needs to be updated in test file

**Review by andrewbranch — APPROVED** [review]

**Comment by muhammeddemir** [comment]
does this also work for vue 3?

**Pull request merged**